File: csharp/MapleStory.OpenAPI/Src/Common/DTO/Character/CharacterBasic.cs
namespace MapleStory.OpenAPI.Common.DTO
{
    public abstract class CharacterBasicDTO
    {
        public abstract DateTimeOffset? Date { get; set; }
        public abstract string CharacterName { get; set; }
        public abstract string WorldName { get; set; }
        public abstract string CharacterGender { get; set; }
        public abstract string CharacterClass { get; set; }
        public abstract string CharacterClassLevel { get; set; }
        public abstract long CharacterLevel { get; set; }
        public abstract long CharacterExp { get; set; }
        public abstract string CharacterExpRate { get; set; }
        public abstract string? CharacterGuildName { get; set; }
        public abstract string CharacterImage { get; set; }
        public abstract DateTimeOffset CharacterDateCreate { get; set; }
        public abstract bool AccessFlag { get; }
        public abstract bool LiberationQuestClearFlag { get; }
    }
}

File: js/src/maplestory/api/common/dto/character/characterBasic.ts
export abstract class CharacterBasicDto {
  public abstract date: Date | null;
  public abstract characterName: string;
  public abstract worldName: string;
  public abstract characterGender: string;
  public abstract characterClass: string;
  public abstract characterClassLevel: string;
  public abstract characterLevel: number;
  public abstract characterExp: number;
  public abstract characterExpRate: string;
  public abstract characterGuildName: string | null;
  public abstract characterImage: string;
  public abstract characterDateCreate: Date | null;
  public abstract accessFlag: 'true' | 'false';
  public abstract liberationQuestClear: string;

  public get isAccessFlag() {
    return this.accessFlag === 'true';
  }
}
